rtic-sync/src/channel.rs

```diff
@@ -77,37 +88,49 @@ impl<T, const N: usize> Channel<T, N> {
         }
     }
 
+    /// Create a new channel.
+    #[cfg(loom)]
+    pub fn new() -> Self {
+        Self {
+            freeq: UnsafeCell::new(Deque::new()),
+            readyq: UnsafeCell::new(Deque::new()),
+            receiver_waker: WakerRegistration::new(),
+            slots: core::array::from_fn(|_| UnsafeCell::new(MaybeUninit::uninit())),
+            wait_queue: WaitQueue::new(),
+            receiver_dropped: UnsafeCell::new(false),
+            num_senders: UnsafeCell::new(0),
+        }
+    }
+
     /// Split the queue into a `Sender`/`Receiver` pair.
     pub fn split(&mut self) -> (Sender<'_, T, N>, Receiver<'_, T, N>) {
+        // SAFETY: we have exclusive access to `self`.
+        let freeq = self.freeq.get_mut();
+        let freeq = unsafe { freeq.deref() };
```

Comment:
This should not need unsafe, something is wrong with the interface here?

Reply:
Sadly, [`loom::cell::UnsafeCell`][0] doesn't have a way to go from mutable-ref/owned `UnsafeCell<T>` to `&mut T` safely. You have to go through `MutPtr<T>`, which you can obtain through an immutably borrowed `UnsafeCell<T>`.

[0]: https://docs.rs/loom/latest/loom/cell/struct.UnsafeCell.html

Reply:
Have moved the safety check to `unsafecell::UnsafeCell::as_mut` and re-worked the exporting a little.